compress.cc

```diff
@@ -990,8 +961,80 @@ class sstable_compressor_factory_impl : public sstable_compressor_factory, publi
     }
 };
 
+default_sstable_compressor_factory::default_sstable_compressor_factory(config cfg)
+    : _cfg(std::move(cfg))
+    , _impl(std::make_unique<sstable_compressor_factory_impl>(_cfg))
+{
+    for (shard_id i = 0; i < smp::count; ++i) {
+        auto numa_id = _cfg.numa_config[i];
+        _numa_groups.resize(std::max<size_t>(_numa_groups.size(), numa_id + 1));
+        _numa_groups[numa_id].push_back(i);
+    }
+}
+
+default_sstable_compressor_factory::~default_sstable_compressor_factory() {
+}
+
+auto default_sstable_compressor_factory_config::from_db_config(
+    const db::config& cfg,
+    std::span<const unsigned> numa_config) -> self
```

Comment:
How does this work? Is `self` a reserved keyword now? Or is there some `tyepedef`/`using` I don't see?

Reply:
It's a `using`.

```c++
struct default_sstable_compressor_factory_config {
    using self = default_sstable_compressor_factory_config;
...
```

Reply:
thanks